org.civicrm.reqangular/src/common/decorators/angular-date/datepicker-addon.js

```diff
@@ -1,35 +1,88 @@
-define([], function () {
+/* eslint-env amd */
+
+define([
+  'common/moment'
+], function (moment) {
   'use strict';
 
   /**
    * Decorates the uib-datepicker-popup directive, making its sibling '.input-group-addon'
    * trigger the datepicker
    */
-  return ['$delegate', function ($delegate) {
+  return ['$delegate', '$compile', function ($delegate, $compile) {
     var directive = $delegate[0];
+    var isMobile = document.getElementsByTagName('body')[0].classList.contains('mobile');
     var origCompile = directive.compile;
+    var calendarIcon = '<span ' +
+      'class="input-group-addon pointer">' +
+      '<i class="fa fa-calendar"></i>' +
+      '</span>';
 
-    directive.compile = function compile() {
-      var link = origCompile.apply(this, arguments);
+    if (isMobile) {
+      directive.require = 'ngModel';
+      directive.compile = function compile () {
+        return function (scope, element, attrs, ngModel) {
+          element.parent('.input-group').css('display', 'block');
+          element.prop('type', 'date');
+          attrs.ngReadonly = false;
 
-      return function (scope, element) {
-        link.apply(this, arguments);
+          handleCalendarIconandEvent(scope, element, attrs);
 
-        function onClick() {
-          element.trigger('click');
-        }
+          ngModel.$formatters.push(function (viewValue) {
+            return convertToHTML5DatepickerFormat(viewValue);
+          });
+
+          scope.$watch('datepickerOptions.minDate', function (value) {
+            attrs.$set('min', convertToHTML5DatepickerFormat(value));
+          });
+          scope.$watch('datepickerOptions.maxDate', function (value) {
+            attrs.$set('max', convertToHTML5DatepickerFormat(value));
+          });
+        };
+      };
+    } else {
+      directive.compile = function compile () {
+        var link = origCompile.apply(this, arguments);
+
+        return function (scope, element, attrs) {
+          handleCalendarIconandEvent(scope, element, attrs);
```

Comment:
`handleCalendarIconAndEvent`

Reply:
After re organizing the code, it does not exist anymore.